Frameworks/CoreLocation/CLLocation.mm

```diff
@@ -228,11 +224,33 @@ - (id)copyWithZone:(NSZone*)zone {
 }
 
 /**
- @Status Stub
+ @Status Interoperable
+ @Notes  This uses the "haversine" formula to find circular distance between two points, found
+         here: https://en.wikipedia.org/wiki/Haversine_formula
 */
 - (CLLocationDistance)distanceFromLocation:(const CLLocation*)location {
-    UNIMPLEMENTED();
-    return StubReturn();
+    // If either location is invalid return -1
+    if (location == nullptr) {
+        return -1.0;
+    }
```

Comment:
thanks; can you update the comment to reflect that you're following reference platform behavior here?


---
In reply to: [71441657](https://github.com/Microsoft/WinObjC/pull/635#discussion_r71441657) [](ancestors = 71441657)